pkg/spireentry/reconciler.go

```diff
@@ -250,9 +258,46 @@ func (r *entryReconciler) recalculateUnsupportFields(ctx context.Context, log lo
 	r.nextGetUnsupportedFields = time.Now().Add(10 * time.Minute)
 }
 
-func (r *entryReconciler) listEntries(ctx context.Context) ([]spireapi.Entry, error) {
+func (r *entryReconciler) shouldProcessOrDeleteEntryID(entry spireapi.Entry) (bool, bool) {
+	if r.config.EntryIDPrefix == "" {
+		return true, false
+	}
+	if strings.HasPrefix(entry.ID, r.config.EntryIDPrefix) {
+		return true, false
+	}
+	if r.config.EntryIDPrefixCleanup != nil {
+		cleanupPrefix := *r.config.EntryIDPrefixCleanup
+		if cleanupPrefix == "" {
+			if strings.Contains(entry.ID, "/") {
+				return false, false
+			}
+			return false, true
+		}
+		if strings.HasPrefix(entry.ID, cleanupPrefix) {
+			return false, true
+		}
+	}
+	return false, false
+}
+
+func (r *entryReconciler) listEntries(ctx context.Context) ([]spireapi.Entry, []spireapi.Entry, error) {
 	// TODO: cache?
-	return r.config.EntryClient.ListEntries(ctx)
+	deleteOnlyEntries := make([]spireapi.Entry, 0)
+	currentEntries := make([]spireapi.Entry, 0)
```

Comment:
nit: i don't think there is value to a zero sized allocation (we don't need these to be non-nil if empty)
```suggestion
	var deleteOnlyEntries []spireapi.Entry
	var currentEntries []spireapi.Entry
```